### PR TITLE
Bootstrap module. Closes #159.

### LINKED ISSUE #159: Add bootstrap module to contain startup logic
## Description
As a developer I want to have a module to contain the boostrap up logic for the server

## Acceptance criteria

- [ ] Module to contain boostrap logic e.g. creating an ecoverse if there is not one already

### PR DESCRIPTION
**Pull request by valentinyanakiev**
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated
- [x] I've read the contributing document https://github.com/cherrytwist/.github/blob/master/CONTRIBUTING.md
- [x] I've read and understand the Code of Conduct https://github.com/cherrytwist/.github/blob/master/CODE_OF_CONDUCT.md
- [x] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/cherrytwist/Coordination/blob/master/LICENSE 
     and the contributor license agreement: tba

### CODE REVIEW

File: src/domain/ecoverse/ecoverse.service.ts [hunk 1]
```diff
@@ -258,6 +258,19 @@ export class EcoverseService {
     return user;
   }
 
+  async addAdmin(user: IUser): Promise<IUser> {
+    const ctverse = await this.getEcoverse();
+    const adminsGroup = await this.userGroupService.getGroupByName(
+      ctverse,
+      'admins'
```
Comment (techsmyth) [inline src/domain/ecoverse/ecoverse.service.ts]: group name should be gotten from Enum for safety

File: src/utils/bootstrap/bootstrap.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { BootstrapService } from './bootstrap.service';
+
+describe('BootstrapService', () => {
```
Comment (techsmyth) [inline src/utils/bootstrap/bootstrap.service.spec.ts:4]: Can you explain the purpose of this class?

File: src/utils/bootstrap/bootstrap.service.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    private ecoverseService: EcoverseService,
+    private userService: UserService
+  ) {}
+
+  async boostrapEcoverse() {
+    this.ecoverseService.getEcoverse();
+    const ctAdminDto = new UserInput();
+    ctAdminDto.name = 'ctAdmin';
+    ctAdminDto.email = 'admin@cherrytwist.org';
+    ctAdminDto.lastName = 'admin';
```
Comment (techsmyth) [inline src/utils/bootstrap/bootstrap.service.ts]: Needs to handle the case where the user already exists and also if the user is already in the admins group. It is a failsafe to make sure this user is there.

Also, need to take the setting of the admin role into the context of authorisation design - how to set the admin profile details? For later...

**Review by techsmyth — CHANGES_REQUESTED** [review]
Looks good - a couple minor ones and some logic also to be added to deal with the user already existing and/or already is an admin.

File: src/utils/bootstrap/bootstrap.service.spec.ts (hunk 2 above)
Comment (valentinyanakiev) [reply src/utils/bootstrap/bootstrap.service.spec.ts:4]: Test harness auto-generated by nestjs.

File: src/utils/bootstrap/bootstrap.service.ts (hunk 3 above)
Comment (valentinyanakiev) [reply src/utils/bootstrap/bootstrap.service.ts]: Bootstrap service calls UserService for user creation - the failsafe is there already. Same for checking whether the user is already in a group - EootstrapService calls EcoverseService which in line calls UserGroup service - and it checks whether the user belongs to a group. Authorisation design - separate User Story(ies).

File: src/domain/ecoverse/ecoverse.service.ts (hunk 1 above)
Comment (valentinyanakiev) [reply src/domain/ecoverse/ecoverse.service.ts]: Agreed and resolved.

File: src/utils/bootstrap/bootstrap.service.ts (hunk 3 above)
Comment (valentinyanakiev) [reply src/utils/bootstrap/bootstrap.service.ts]: I saw the failure and addressed it.

File: src/utils/bootstrap/bootstrap.service.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  async boostrapEcoverse() {
+    try {
+      this.ecoverseService.getEcoverse();
+      const ctAdminDto = new UserInput();
+      ctAdminDto.name = 'ctAdmin';
+      ctAdminDto.email = 'admin@cherrytwist.org';
+      ctAdminDto.lastName = 'admin';
+
+      if (this.userService.getUserByEmail('admin@cherrytwist.org')) {
```
Comment (techsmyth) [inline src/utils/bootstrap/bootstrap.service.ts]: Afraid logic does not look correct...if the user already exists you still want the person to be assigned as an admin

**Review by techsmyth — APPROVED** [review]

**Pull request merged**